enhancements/vdpa/virtio-vdpa-ovn-hw-offload.md

```diff
@@ -0,0 +1,237 @@
+
+## Summary
+See [VDPA support in OCP (Overview)](https://github.com/openshift/enhancements/blob/master/enhancements/vdpa/vdpa-support-overview.md).
+
+Main aspects of this enhancement:
+- primary interface in the pod is configured as a Virtio/vDPA device
+- NIC configured in switchdev mode
+- OVS HW offloading enabled
+- Supported HW: NVIDIA CTX-6 DX
+
+
+## Motivation
+
+The main motivation is to integrate a Virtio/vDPA device in Openshift.
+
+
+### User Stories
+
+Details are provided in the overview enhancement.
+
+### Goals
+
+- As a user I want to configure an Openshift cluster with Virtio/VDPA support.
+- As a user I want to determine how many VFs to create for a given NIC and how each VF should be consumed by workloads (e.g., netdev device in the container network namespace)
+- As a user I want the VDPA solution to be integrated with OVN-kubernetes and with HW offloading enabled
+
+### Non-Goals
+
+- The Vhost-vdpa driver won't be supported in CNFs (VMs only). The virtio-vdpa driver will be supported in CNFs, i.e. netdev device to be consumed.
+- Scalable IOV and Sub-functions are out of scope, only switchdev SR-IOV will be supported (with OVS HW offloading to the NIC)
+
+## Proposal
+
+The following diagram depicts all the involved components and their interactions:
+
+![alt text](vdpa-architecture.png "VDPA architecture")
+
+### Workflow Description
+
+#### Pre-requirements
+
+- Create the Openshift cluster on a baremetal node
+- Deploy OVN-kubernetes
+- Deploy the SRIOV network operator
+
+#### SRIOV network operator workflow
+
+- Configure OpenVswitch in HW offload mode
+- Configure NIC (single script?)
+  - Install vendor-independent kernel drivers: vdpa, virtio-vdpa
+  - Configure VFs (same as SR-IOV: echo N > /sys/devices/{PF}/sriov_numvfs)
+  - Put NIC in switchdev mode (smart NIC)
+  - Enable HW offload mode on PF, VFs and port representors
+  - Disable Network Manager on PF, VFs and port representors
+  - Install vendor-dependent kernel drivers (e.g. mlx5-core for Mellanox cards)
+  - Bind the right PCI driver (vendor specific). Some vendors might implement vdpa in a different PCI driver (e.g: Intel’s ifcvf). Others might keep the same pci driver and require extra steps (e.g: in mellanox vdpa, the VF is still bound to “mlx5_core”)
+- Create the vdpa device (vendor specific). Some vendors might require no extra steps because they create the vdpa device on a pci driver probe (e.g: ifcvf). Others might need extra steps (e.g: Mellanox requires loading an additional driver and in the future, they 	might require managing a “virtual bus”(source)). The plan is to extend the govdpa library to provide this functionality.
+- Bind the vdpa device to the correct vdpa bus driver (virtio-vdpa driver in the first implementation). This is not vendor specific and uses a sysfs-based API
+- Configure the SRIOV device plugin
+  - Add the appropriate vdpa flags to the SR-IOV Device Plugin’s configMap.
+  - Create the Network Attachment Definitions
```

Comment:
`Create the Network Attachment Definitions` could be a separate item, not a sub-item under device plugin.

Reply:
done